### PR TITLE
Adding support for fft project

### PR DESCRIPTION
**Pull request by chick**
Mostly bringing some new special flags to do historgram analysis of wires
Report of wire usage is written out to files
New options control where to write them
NumberMonitor does the analysis

### CODE REVIEW

File: src/test/scala/firrtl_interpreter/NumberMonitorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      monitor.update(value)
+    }
+
+    println(s"mean = ${monitor.mean}, σ = ${monitor.stddev} min = ${monitor.minValue} max = ${monitor.maxValue}")
+    println(s"bins = ${monitor.showBins}")
+
+    //    monitor.mean.abs should be (5.0)
+    //    monitor.variance should be (0.0)
+    //    monitor.minValue should be (42.0)
+    //    monitor.maxValue should be (42.0)
```
Comment (ucbjrl) [inline src/test/scala/firrtl_interpreter/NumberMonitorSpec.scala:59]: Are these useful?

File: src/test/scala/firrtl_interpreter/NumberMonitorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
```
Comment (ucbjrl) [inline src/test/scala/firrtl_interpreter/NumberMonitorSpec.scala:80]: Are these useful?

File: src/test/scala/firrtl_interpreter/NumberMonitorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+    monitor.bins.last should be (1)
+
+    monitor.update(lower * BigInt(1000))  // illegal low value treated as low
+    monitor.update(upper * BigInt(1000))  // illegal high value treated as high
+
+    monitor.bins.head should be (2)
+    monitor.bins.last should be (2)
+
+    // println(s"Bins ${monitor.bins.mkString(",")}")
```
Comment (ucbjrl) [inline src/test/scala/firrtl_interpreter/NumberMonitorSpec.scala:170]: scalpel left in patient?

**Comment by chick** [comment]
@ucbjrl Removed the offending tests

**Pull request merged**